### DISCUSSION TITLE
Require authentication to GraphQL and StoreFront ?

### DISCUSSION THREAD

**Discussion by anesfassih** [saleor/saleor#7943] [Q&A]
Hello everyone, I discover the incredible project that is Saleor, and I wonder if it is possible to use it for a B2B platform, I believe it is possible, but I have some unanswered questions in mind, and I would like to have the opinion of someone who knows the project well! And the main question is about a strange need for an e-commerce platform : Is it possible to give access to the platform only to authenticated users? This concerns both the GraphQl Queries and the StoreFront. And if it's possible how to do it (in outline of course).

**Comment by karolkielecki** [comment]
@anesfassih Hi! I think that would be possible, but please provide more information. Could you please tell us a use case?

**Reply by anesfassih** [reply]
To begin, I would like to thank you for your time and contribution to the project @karolkielecki !

I will explain in more detail the needs we have:
We need to give access to product information only to actors that we have authorized through the activation of an account after registration. So, access via GraphQL must be denied for data (like products, categories, etc...), and on the front-end side, the user will have access to a landing page and presentation pages...  And of course a registration form with more information. And if he is authenticated he will be able to browse products, categories, checkout and order features.

**Reply by anesfassih** [reply]
?

**Reply by karolkielecki** [reply]
Hey. So it's a more complicated use-case than we thought. Maybe you can use our permission system. But I think you'll have to create a middleware with a specific logic. Without modifying the core, it will be difficult to achieve it. We do not recommend modifying the code because it will be challenging to stay up to date with the next releases.

**Reply by anesfassih** [reply]
Thanks for the feedback @karolkielecki. We will take this in consideration.
